class-instant-articles-publisher.php

```diff
@@ -93,10 +98,12 @@ public static function submit_article( $post_id, $post ) {
 
 				try {
 					// Import the article.
-					$client->importArticle( $article, $take_live );
+					$submission_id = $client->importArticle( $article, $take_live );
+					update_post_meta( $post_id, self::SUBMISSION_ID_KEY, $submission_id );
 				} catch ( Exception $e ) {
 					// Try without taking live for pages not yet reviewed.
-					$client->importArticle( $article, false );
+					$submission_id = $client->importArticle( $article, false );
+					update_post_meta( $post_id, self::SUBMISSION_ID_KEY, $submission_id );
```

Comment:
When we try to publish, we send it with the flag to take it live. If the page is still pending review, this particular call will fail and an Exception is thrown. We then try to send it without the flag.

When the page is under review, we always send the request twice. But when the page is approved, we always send a single request.

We could, instead, check the review status before and include the flag or not based on the status. But that would mean 2 requests are made (checking the status then sending) in both scenarios. It is a clearer solution, but the current solution performs better on the average case.

We can discuss it later, it is not on the scope of this PR.